src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs

```diff
@@ -138,27 +159,31 @@ public static class PackageDownloader
                 throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Strings.DownloadResourceNotFound, sourceRepository.PackageSource.Source));
             }
 
-            var downloadResourceResult
+            DownloadResourceResult result = null;
+
+            token.ThrowIfCancellationRequested();
+
+            result
                 = await downloadResource.GetDownloadResourceResultAsync(packageIdentity, settings, token);
 
-            if (downloadResourceResult == null)
+            if (result == null)
             {
-                throw new InvalidOperationException(string.Format(
+                throw new FatalProtocolException(string.Format(
```

Comment:
Changing the exception type here my affect how PS, VS UI, and VS API handle this.